hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/storefiletracker/FileBasedStoreFileTracker.java

```diff
@@ -77,7 +77,7 @@ protected List<StoreFileInfo> doLoadStoreFiles(boolean readOnly) throws IOExcept
     for (StoreFileEntry entry : list.getStoreFileList()) {
       infos.add(ServerRegionReplicaUtil.getStoreFileInfo(conf, fs, ctx.getRegionInfo(),
         ctx.getRegionFileSystem().getRegionInfoForFS(), ctx.getFamily().getNameAsString(),
-        new Path(ctx.getFamilyStoreDirectoryPath(), entry.getName())));
+        new Path(ctx.getFamilyStoreDirectoryPath(), entry.getName()), this));
```

Comment:
>So the plan is to refactor the reference file handling into StoreFileTrackerBase, and simply inherit it here for now and otherwise do not change it; and then on some later JIRA maybe evolve the representation and handling references with overrides in FileBasedStoreFileTracker somehow. Correct? That's why we are moving more into SFT? :-)
Yes

>Consider writing up a couple of sentences about the motivation of this work and the long term directions and putting that somewhere near the top of the PR description?
Sure